src/linear_algebra/matrix/schur_complement.lean

```diff
@@ -54,6 +58,155 @@ lemma from_blocks_eq_of_invertible₂₂
+  convert (rfl : ⅟(from_blocks A B 0 D) = _),
+end
+
+lemma inv_of_from_blocks_zero₁₂_eq
+  (A : matrix m m α) (C : matrix n m α) (D : matrix n n α)
+  [invertible A] [invertible D] [invertible (from_blocks A 0 C D)] :
+  ⅟(from_blocks A 0 C D) = from_blocks (⅟A) 0 (-(⅟D⬝C⬝⅟A)) (⅟D) :=
+begin
+  letI := from_blocks_zero₁₂_invertible A C D,
+  convert (rfl : ⅟(from_blocks A 0 C D) = _),
+end
+
+/-- Both diagonal entries of an invertible upper-block-triangular matrix are invertible (by reading
+off the diagonal entries of the inverse). -/
+def invertible_of_from_blocks_zero₂₁_invertible
+  (A : matrix m m α) (B : matrix m n α) (D : matrix n n α)
+  [invertible (from_blocks A B 0 D)] : invertible A × invertible D :=
+{ fst := invertible_of_left_inverse _ (⅟(from_blocks A B 0 D)).to_blocks₁₁ $ begin
+    have := matrix.inv_of_mul_self (from_blocks A B 0 D),
+    rw [←from_blocks_to_blocks (⅟(from_blocks A B 0 D)), from_blocks_multiply] at this,
+    replace := congr_arg matrix.to_blocks₁₁ this,
+    simpa only [matrix.to_blocks_from_blocks₁₁, matrix.mul_zero, add_zero, ←from_blocks_one]
+      using this,
+  end,
+  snd := invertible_of_right_inverse _ (⅟(from_blocks A B 0 D)).to_blocks₂₂ $ begin
+    have := matrix.mul_inv_of_self (from_blocks A B 0 D),
+    rw [←from_blocks_to_blocks (⅟(from_blocks A B 0 D)), from_blocks_multiply] at this,
+    replace := congr_arg matrix.to_blocks₂₂ this,
+    simpa only [matrix.to_blocks_from_blocks₂₂, matrix.zero_mul, zero_add, ←from_blocks_one]
+      using this,
+  end }
+
+/-- Both diagonal entries of an invertible lower-block-triangular matrix are invertible (by reading
+off the diagonal entries of the inverse). -/
+def invertible_of_from_blocks_zero₁₂_invertible
+  (A : matrix m m α) (C : matrix n m α) (D : matrix n n α)
+  [invertible (from_blocks A 0 C D)] : invertible A × invertible D :=
+{ fst := invertible_of_right_inverse _ (⅟(from_blocks A 0 C D)).to_blocks₁₁ $ begin
+    have := matrix.mul_inv_of_self (from_blocks A 0 C  D),
+    rw [←from_blocks_to_blocks (⅟(from_blocks A 0 C  D)), from_blocks_multiply] at this,
+    replace := congr_arg matrix.to_blocks₁₁ this,
+    simpa only [matrix.to_blocks_from_blocks₁₁, matrix.zero_mul, add_zero, ←from_blocks_one]
+      using this,
+  end,
+  snd := invertible_of_left_inverse _ (⅟(from_blocks A 0 C D)).to_blocks₂₂ $ begin
+    have := matrix.inv_of_mul_self (from_blocks A 0 C  D),
+    rw [←from_blocks_to_blocks (⅟(from_blocks A 0 C D)), from_blocks_multiply] at this,
+    replace := congr_arg matrix.to_blocks₂₂ this,
+    simpa only [matrix.to_blocks_from_blocks₂₂, matrix.mul_zero, zero_add, ←from_blocks_one]
+      using this,
+  end }
+
+/-- `invertible_of_from_blocks_zero₂₁_invertible` and `from_blocks_zero₂₁_invertible` form
+an equivalence. -/
+def from_blocks_zero₂₁_invertible_equiv (A : matrix m m α) (B : matrix m n α) (D : matrix n n α) :
+  invertible (from_blocks A B 0 D) ≃ invertible A × invertible D :=
```

Comment:
I suppose `iff` wouldn't work here because the two sides aren't `Prop`-valued? Showing an `equiv` between two subsingletons seems overly complicated...